src/libaktualizr/package_manager/ostreemanager.cc

```diff
@@ -18,6 +18,13 @@
 
 static void aktualizr_progress_cb(OstreeAsyncProgress *progress, gpointer data) {
   auto *mt = static_cast<PullMetaStruct *>(data);
+  bool has_been_paused = false;
+  if (!mt->pause_mutex->try_lock()) {
+    mt->pause_mutex->lock();
+    has_been_paused = true;
+  } else {
+    mt->pause_mutex->unlock();
+  }
```

Comment:
I'm not sure I entirely understand this logic. So if the mutex is already locked, we want to acquire it again? I assume that is just to wait until it is freed/unpaused. But then why wait until the end of this function before unlocking it again?

Whatever the intention, a comment would be helpful here. This logic is tricky enough that it is worth stating the goals explicitly.

Reply:
Updated with comments